model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -165,6 +165,19 @@ class ModelReplicationServer(val repositoriesManager: RepositoriesManager) {
             call.respondDelta(initialVersion.getContentHash(), null)
         }
 
+        post<Paths.deleteRepository> {
+            fun ApplicationCall.repositoryId() = RepositoryId(parameters["repository"]!!)
+            fun PipelineContext<Unit, ApplicationCall>.repositoryId() = call.repositoryId()
+
+            val repositoryId = repositoryId()
+            if (!repositoriesManager.repositoryExists(repositoryId)) {
+                call.respond(HttpStatusCode.NotFound)
+            } else {
+                repositoriesManager.removeRepository(repositoryId)
+                call.respond(HttpStatusCode.NoContent)
+            }
```

Comment:
One more thing I didn't notice before: This is probably not deterministic. Between the test for `repositoryExists` and the `removeRepositoryCall`, some other call might have already removed the repo. This implementation here would need to ensure that the 404 response is also sent in that case.

Reply:
I moved those operations into the transaction to make them atomic. Fixed in [dc41ece](https://github.com/modelix/modelix.core/pull/450/commits/dc41ece986a8ee4fc698762d99dc5163ec832336).